src/main/java/com/microsoft/azure/relay/AsyncSemaphore.java

```diff
@@ -33,19 +36,19 @@ int availablePermits() {
 		}
 	}
 
-	CompletableFuture<LockRelease> acquireAsync(AutoShutdownScheduledExecutor executor) {
```

Comment:
>CompletableFuture<LockRelease> acquireAsync [](start = 1, length = 43)

maybe we should make these acquire methods public

Reply:
They definitely should be. I'm a bit unclear if the decided names are supposed to be acquireAsync or lockAsync since both occur.

---
In reply to: [259560190](https://github.com/Azure/azure-relay-java/pull/22#discussion_r259560190) [](ancestors = 259560190)

Reply:
I would prefer acquireAsync to be similar to the Java API

---
In reply to: [259912209](https://github.com/Azure/azure-relay-java/pull/22#discussion_r259912209) [](ancestors = 259912209,259560190)